www/includes/easyparliament/member.php

```diff
@@ -242,6 +242,17 @@ public function date_in_memberships($house, $date){
         return false;
     }
 
+    public function member_agreements($policyID, $house = HOUSE_TYPE_COMMONS){
+        // agreements that a member has been present for on a specific policy
+        $policiesList = new MySociety\TheyWorkForYou\Policies($policyID);
+        $rel_agreements = $policiesList->all_policy_agreements[$policyID] ?? [];
+        $MEMBER = $this;
```

Comment:
More out of interest, is this needed, does `use ($this...` not work okay?

Reply:
I got an error for it - this looks like the documentation for that. Nicer approaches welcome! https://www.php.net/manual/en/functions.anonymous.php#:~:text=Any%20such%20variables%20must%20be%20passed%20to%20the%20use%20language%20construct.%20As%20of%20PHP%207.1%2C%20these%20variables%20must%20not%20include%20superglobals%2C%20%24this%2C%20or%20variables%20with%20the%20same%20name%20as%20a%20parameter

Reply:
I think $this is automatically available through the closure without `use`, e.g.
`php -r'class Foo { public $allow = 2; function foo() { $a = [1,2,3,4]; $a = array_filter($a, function($n) { return $n === $this->allow; }); return $a; } }; print_r( (new Foo)->foo() );'`
Outputs 2.

So hopefully:
```
        $rel_agreements = array_filter($rel_agreements, function($agreement) use ($house) {
            return $this->date_in_memberships($house, $agreement['date']);
        });
```
would work.

Reply:
Yep, that works as expected!